### PR TITLE
Support "stub_distribution" and "upload" metadata fields

### PR DESCRIPTION
**Pull request by srittau**

### CODE REVIEW

File: tests/check_consistent.py [hunk 1]
```diff
@@ -40,7 +51,7 @@ def assert_consistent_filetypes(
         if entry.is_file():
             if not allow_nonidentifier_filenames:
                 assert entry.stem.isidentifier(), f'Files must be valid modules, got: "{entry}"'
-                bad_filetype = f'Only {extension_descriptions[kind]!r} files allowed in the "{directory}" directory; got: {entry}'
+            bad_filetype = f'Only {extension_descriptions[kind]!r} files allowed in the "{directory}" directory; got: {entry}'
```
Comment (AlexWaygood) [inline tests/check_consistent.py:54]: Thanks for fixing this :)

File: tests/check_consistent.py [hunk 2]
```diff
@@ -150,6 +161,11 @@ def check_metadata() -> None:
             # Check that the requirement parses
             Requirement(dep)
 
+        if "stub_distribution" in data:
+            assert dist_name_re.fullmatch(data["stub_distribution"]), f"Invalid 'stub_distribution' value for {distribution}"
```
Comment (AlexWaygood) [inline tests/check_consistent.py:165]:
```suggestion
            assert dist_name_re.fullmatch(data["stub_distribution"]), f"Invalid 'stub_distribution' value for {distribution!r}"
```

File: tests/check_consistent.py [hunk 3]
```diff
@@ -150,6 +161,11 @@ def check_metadata() -> None:
...
+
+        assert isinstance(data.get("upload", True), bool), f"Invalid 'upload' value for {distribution}"
```
Comment (AlexWaygood) [inline tests/check_consistent.py:167]:
```suggestion
        assert isinstance(data.get("upload", True), bool), f"Invalid 'upload' value for {distribution!r}"
```

**Review by AlexWaygood — APPROVED** [review]

**Comment by srittau** [comment]
Also, GitHub Copilot is frightening sometimes:

copilot

**Comment by srittau** [comment]
I'm ignoring the unrelated pyright errors.

**Pull request merged**